app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/messages/MessageDetailViewModel.java

```diff
@@ -0,0 +1,40 @@
+package com.teamagam.gimelgimel.app.model.ViewsModels.messages;
+
+import com.teamagam.gimelgimel.app.common.DataChangedObservable;
+import com.teamagam.gimelgimel.app.model.ViewsModels.Message;
+import com.teamagam.gimelgimel.app.model.entities.messages.SelectedMessageModel;
+
+/**
+ * Shared functionality of detail message view-model
+ * Verifies selected message is of appropriate type before each method call.
+ * Notifies observers on data changes only when selected message is of appropriate type.
+ */
+abstract class MessageDetailViewModel extends SelectedMessageViewModel implements DataChangedObservable {
```

Comment:
why it is not public?
I would like to hold reference to MessageDetailViewModel so i would not need to update the title every time.